src/bnb_trading/historical_tester.py

```
@@ -221,44 +237,58 @@
-                logger.info(f"Филтрирани weekly данни: {len(weekly_df)} редове за периода {start_date} до {end_date}")
+                weekly_df = weekly_df[
+                    (weekly_df.index >= weekly_start) & (weekly_df.index <= weekly_end)
+                ]
+                data["weekly"] = weekly_df
+                logger.info(
+                    f"Филтрирани weekly данни: {
+                        len(weekly_df)} редове за периода {start_date} до {end_date}"
+                )
 
         except Exception as e:
             logger.error(f"❌ Грешка при fetch на данни: {e}")
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)